File: eTheater.Services/Database/Ticket.cs
using System;
using System.Collections.Generic;

namespace eTheater.Services.Database;

public partial class Ticket : BaseEntity
{
    public int TicketId { get; set; }

    public string? Seat { get; set; }

    public string? NumberOfRow { get; set; }

    public int? NumberOfSeat { get; set; }

    public bool IsActive { get; set; } = true;

    public int? ShowScheduleId { get; set; }

    public int? PurchaseId { get; set; }

    public virtual Purchase? Purchase { get; set; }

    public virtual ShowSchedule? ShowSchedule { get; set; }
}

File: UI/etheater_web/lib/models/ticket.dart
import 'package:json_annotation/json_annotation.dart';
import 'models.dart';

part 'ticket.g.dart';

@JsonSerializable()
class Ticket {
  int ticketId;
  String seat;
  String numberOfRow;
  int numberOfSeat;
  bool isActive;
  int? showScheduleId;
  ShowSchedule? showSchedule;
  int? purchaseId;
  Purchase? purchase;
  bool? isDeleted;

  Ticket(
      {required this.ticketId,
      required this.seat,
      required this.numberOfRow,
      required this.numberOfSeat,
      required this.isActive,
      this.showScheduleId,
      this.showSchedule,
      this.purchaseId,
      this.purchase,
      this.isDeleted});

  factory Ticket.fromJson(Map<String, dynamic> json) => _$TicketFromJson(json);
  Map<String, dynamic> toJson() => _$TicketToJson(this);
}
